client/src/components/profile/profile.tsx

```diff
@@ -112,10 +90,12 @@ function renderProfile(user: ProfileProps['user']): JSX.Element {
 function Profile({ user, isSessionUser }: ProfileProps): JSX.Element {
   const { t } = useTranslation();
   const {
-    profileUI: { isLocked = true },
+    profileUI: { isLocked },
     username
   } = user;
 
+  const isPublicProfile = !isLocked || isSessionUser;
```

Comment:
```suggestion
  const showUserProfile = !isLocked || isSessionUser;
```

Call it what it does.